pkg/controller/server_status_request_controller_test.go

```diff
@@ -0,0 +1,270 @@
+/*
+Copyright 2020 the Velero contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controller
+
+import (
+	"context"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/clock"
+	"k8s.io/client-go/kubernetes/scheme"
+	ctrl "sigs.k8s.io/controller-runtime"
+	kbclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+
+	"github.com/vmware-tanzu/velero/internal/velero"
+	velerov1api "github.com/vmware-tanzu/velero/pkg/apis/velero/v1"
+	"github.com/vmware-tanzu/velero/pkg/builder"
+	"github.com/vmware-tanzu/velero/pkg/buildinfo"
+	"github.com/vmware-tanzu/velero/pkg/plugin/framework"
+	velerotest "github.com/vmware-tanzu/velero/pkg/test"
+)
+
+func statusRequestBuilder(resourceVersion string) *builder.ServerStatusRequestBuilder {
+	return builder.ForServerStatusRequest(velerov1api.DefaultNamespace, "sr-1", resourceVersion)
+}
+
+var _ = Describe("Server Status Request Reconciler", func() {
+	BeforeEach(func() {})
+	AfterEach(func() {})
+
+	It("Should successfully patch a server status request object status phase", func() {
+		// now will be used to set the fake clock's time; capture
+		// it here so it can be referenced in the test case defs.
+		now, err := time.Parse(time.RFC1123, time.RFC1123)
+		Expect(err).To(BeNil())
+		now = now.Local()
+
+		tests := []struct {
```

Comment:
suggest adding a `name` field to this testCase struct to use for naming testcases. Comments in each testCase struct instance can be the `name` for the testCase.
This field can also be used to report test failures.

Reply:
We are not using the `testing.T` package for this test, we are using ginkgo and gomega. So, it's not possible to do `t.Run(tt.name, func(t *testing.T) {`.

I don't know of any other way to inject the name into test failures. And if we are not using the name in test failures, I don't think we should have a `name` variable but instead the comments are applicable here.

Opinions?

Reply:
Since we're mixing patterns here (BDD and table-driven tests), I think we'd want to look at [Ginkgo's table driven test extension](https://onsi.github.io/ginkgo/#table-driven-tests) which provides an `Entry` construct for labeling the tests.

We can also use [`It`](https://onsi.github.io/ginkgo/#individual-specs-it) blocks for naming individual cases.

I think either of these would get us printable output, which would be helpful in debugging.

Admittedly, though, I haven't used Ginkgo in practice, so I'm basing this solely on reading the docs. They don't seem to set up table-driven tests as much, but rather lean on describing individual tests cases in full.

Reply:
I'll add this as something to investigate/add in the "clean up" issue for the kubebuilder stuff, so all controllers get done if there's something to change.

Reply:
In the interest of time, I'm ok with that.